src/App.tsx

```diff
@@ -1,24 +1,177 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { addTodo, getTodos, deleteTodo } from './api/todos';
+import { Header } from './Components/Header';
+import { TodoList } from './Components/TodoList';
+import { Footer } from './Components/Footer';
+import { Notification } from './Components/Notification';
 
-const USER_ID = 0;
+const USER_ID = 7011;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [selectedStatus, setSelectedStatus] = useState<Status>(Status.All);
+  const [isError, setIsError] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDisabledInput, setIsDisabledInput] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isWaitingForDelete, setIsWaitingForDelete] = useState(0);
+  const [isDeletingCompleted, setIsDeletingCompleted] = useState(false);
+
+  const handleNewTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (selectedStatus) {
+        case Status.Active:
+          return !todo.completed;
+        case Status.Completed:
+          return todo.completed;
+        default:
+          return todo;
+      }
+    });
+  }, [selectedStatus, todos]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      })
+      .catch(() => {
+        setIsError(true);
+        setErrorMessage('Unable to add a todo');
+        setTimeout(() => {
+          setIsError(false);
+        }, 3000);
```

Comment:
1. use only 1 state to keep the error 
2. move this piece of code into a function and reuse throughout your code